x-pack/test/functional_basic/apps/ml/permissions/no_ml_access.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+import { USER } from '../../../../functional/services/ml/security_common';
+
+export default function ({ getPageObjects, getService }: FtrProviderContext) {
+  const PageObjects = getPageObjects(['common', 'error']);
+  const ml = getService('ml');
+
+  const testUsers = [USER.ML_UNAUTHORIZED, USER.ML_UNAUTHORIZED_SPACES];
+
+  describe('for user with no ML access', function () {
+    for (const user of testUsers) {
+      describe(`(${user})`, function () {
+        before(async () => {
+          await ml.securityUI.loginAs(user);
+        });
+
+        after(async () => {
+          await ml.securityUI.logout();
+        });
+
+        it('should not allow to access the ML app', async () => {
```

Comment:
Nit - typo - `should not allow access to the ML app`

Reply:
Fixed in c4b15a6